### PR TITLE
PAYEE-683: displays ceased employments after the current tax year

### PR DESCRIPTION
**Pull request by sergi-hmrc**
- [ ] Unit tests passing
- [ ] Coverage reached
- [ ] Acceptance tests passing
- [ ] Reviewed

### CODE REVIEW

File: app/uk/gov/hmrc/tai/viewModels/TaxAccountSummaryViewModel.scala [hunk 1]
```diff
@@ -60,9 +60,19 @@ object TaxAccountSummaryViewModel extends ViewModelHelper {
 
     val pensionsViewModels = incomesSources.livePensionIncomeSources.map(IncomeSourceViewModel.createFromTaxedIncome(_))
 
-    val ceasedEmploymentViewModels = incomesSources.ceasedEmploymentIncomeSources.map(
-      IncomeSourceViewModel.createFromTaxedIncome(_)) ++
-      nonMatchingCeasedEmployments.map(IncomeSourceViewModel.createFromEmployment(_))
+    def employmentCeasedThisYear(employment: Employment): Boolean = {
+      val currentYear = TaxYear()
+      //Default to true as if there is no endDate it's potentially ceased
+      employment.endDate.fold(true)(currentYear.start.minusDays(1).isBefore)
```
[the remark this answers is not in the document]
Comment (sergi-hmrc) [reply app/uk/gov/hmrc/tai/viewModels/TaxAccountSummaryViewModel.scala:66]: x => currentYear.start.minusDays(1) isBefore  x

currentYear.start.minusDays(1).isBefore(_)

currentYear.start.minusDays(1).isBefore

Although maybe it's not expressive enough and i should go for the first one

[the remark this answers is not in the document]
Comment (sergi-hmrc) [reply app/uk/gov/hmrc/tai/viewModels/TaxAccountSummaryViewModel.scala:66]: btw the minus one day is just to include the 6th of April

[the remark this answers is not in the document]
Comment (PerrettJ4) [reply app/uk/gov/hmrc/tai/viewModels/TaxAccountSummaryViewModel.scala:66]: okay with you now, thanks, no need to change it I just didn't know the shorthand

**Review by PerrettJ4 — APPROVED** [review]
looks good

File: app/uk/gov/hmrc/tai/viewModels/TaxAccountSummaryViewModel.scala (hunk 1 above)
[the remark this answers is not in the document]
Comment (sergi-hmrc) [reply app/uk/gov/hmrc/tai/viewModels/TaxAccountSummaryViewModel.scala:66]: Do you think this is more readable? 
!(endDate isBefore currentYear.start)

File: test/uk/gov/hmrc/tai/viewModels/TaxAccountSummaryViewModelSpec.scala [hunk 2]
```diff
@@ -163,6 +163,26 @@ class TaxAccountSummaryViewModelSpec extends BaseSpec with TaxAccountSummaryTest
           sut.ceasedEmployments(1).name mustBe "Employer name4"
         }
       }
+      "has no ceased employment list" when {
+        "employments ceased before the tax year" in {
+          val ceasedEmployments = incomeSources.ceasedEmploymentIncomeSources.map { taxCodeIncome =>
```
[the remark this answers is not in the document]
Comment (sergi-hmrc) [reply test/uk/gov/hmrc/tai/viewModels/TaxAccountSummaryViewModelSpec.scala:168]: Potentially ceased means that it has None as endDate and in the other test there's one employment with some and another one with none so this case should be covered

**Review by brendanstraughan — APPROVED** [review]

**Pull request merged**